### DISCUSSION TITLE
Haskell MOOC - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#383] [Announcements]
# Haskell MOOC - CS自学指南

CS自学指南

https://csdiy.wiki/%E7%BC%96%E7%A8%8B%E5%85%A5%E9%97%A8/Haskell-MOOC/

**Comment by TalkIsCheap22** [comment]
留个纪念，开始努力💪

**Reply by Edwinzheng2323** [reply]
加油加油，一起开始！

**Reply by dy199681** [reply]
冲！

**Reply by starsurplus** [reply]
冲冲冲

**Reply by MengyangGao** [reply]
加油加油，一起努力！

**Reply by clearweater** [reply]
冲冲冲！

**Comment by ghost** [comment]
加油，1/11号我会来正式开始学习的捏！！！

**Reply by Edwinzheng2323** [reply]
1.12开始学习

**Reply by likesnow97** [reply]
2.11开始学习

**Reply by 325146** [reply]
我会在3.8来学习捏！

**Comment by ZHANG-HANBERGER** [comment]
努力，努力。

**Comment by wsChendp** [comment]
开冲

**Comment by Larry-Cui** [comment]
这个博客简洁美观，作者不反对的话，我就抄去用了😃

**Comment by YChenJ** [comment]
20230114大二计算机来看看

**Reply by We-unite** [reply]
兄弟，俺也是CS大二，可以一起学不

**Comment by up-pika** [comment]
2023/1/16 开始挑战

**Comment by shuzhiwei** [comment]
CS全称是啥？

**Reply by starsurplus** [reply]
computer science

**Comment by starsurplus** [comment]
留个纪念 开始努力

**Comment by Yoriskkk** [comment]
而这一切，没有学分，没有绩点，没有老师，没有同学，只有一个信念 —— 你在变强。

**Reply by Yoriskkk** [reply]
这句话真的太tm酷了，工作三年多了，感觉都忘了当初驱使自己前进的初心了

**Comment by showthesunli** [comment]
为什么这么多打鸡血的？

**Comment by leolegend91** [comment]
赞！

**Comment by BinReco** [comment]
2023-01-19 开始

**Comment by killus-tianyi** [comment]
踏进cs大门

**Comment by dengyunsheng250** [comment]
开始努力

**Comment by LetThoughtsFlowOut** [comment]
Let's go!

**Comment by crssss1** [comment]
留念，努力

**Comment by poqpoqpoqpoq** [comment]
emm 稍微做一点补充 文章里提到的 Language Reactor 尽管尚未推出edge的插件 但是edge支持chrome的插件 所以一样可以使用（虽说下个浏览器也不麻烦）

**Comment by Chunlei-L** [comment]
开始学习，冲！

**Comment by Silzzz0717** [comment]
记录一下，2023/01/25

**Comment by godClan** [comment]
Hello, World!

**Comment by NoobOfC** [comment]
1/30開始學習

**Comment by SongYHs** [comment]
1.30开始学习

**Comment by GiseleGong** [comment]
2023年1月31日留下第一个脚印，希望一段时间回来，我是不一样的人了！

**Comment by Al2S36H2O** [comment]
想请问一下大家伙，我看着英文就头疼怎么办呀

**Reply by We-unite** [reply]
我个人用的是[Copytranslator](https://copytranslator.github.io/)，有谷歌/百度/有道等翻译，支持拖拽直接翻译，感觉不错；在本指南中，[日常学习工作流](https://csdiy.wiki/%E5%BF%85%E5%AD%A6%E5%B7%A5%E5%85%B7/workflow/#_7)也有给出其他方法

**Comment by Jin9991** [comment]
感谢大佬。正在开始学shell和重学python。
自己在自学规划里头看了一下，没有找到相关的课所以想问一下各位大佬有推荐的相关课程吗？
想找Computer Theory 和 Computer Organization相关的课程。

**Reply by showthesunli** [reply]
shell 学 Missing Semester，这东西不需要多精通
python 学 cs61a，难的话学 cs50
话说这都是 csdiy 里面提到过的呀，你肯定没仔细看网站

**Reply by MTDickens** [reply]
一年过去了，shell 可以看看 Sysadmin DeCal，比 Missing Semester 对新手友好很多。

**Comment by helang136** [comment]
非计算机专业也想学习

**Comment by DesolateVE** [comment]
谢谢，光看目录就已经被劝退了

**Comment by mobai-ink** [comment]
希望自己能够坚持下来，加油！

**Comment by CaiZhuoyue** [comment]
请问大家一下～CS61B里的gitlet项目可以写到简历上吗～用java实现了git的一些简单操作，比如add,remove,commit和branch这些

**Reply by ProgrammerLMS** [reply]
本科校招还可以，硕士不太行了

**Comment by TephrocactusMYC** [comment]
看了这个才发现一件事儿，我们学校的面向对象用的是C++。

**Reply by MTDickens** [reply]
我们学校用的也是 C++。
不过很多人写 C++ 都写的实际上是 C with C++ STL。如果用 Java 可以强迫使用 OOP，就像 Haskell 强迫使用 FP。当然 C++ 也能教 OOP。

**Comment by MTDickens** [comment]
Haskell 很有意思，建议进行学习。而且建议不要用 UPenn 的 Haskell（在 Reddit 上，上过 UPenn 课程的说，UPenn 的课比这个课程简单）

**Reply by curbe454** [reply]
然而在推荐资料上 UPenn 的 CIS194 做得更好，它鼓励人去看 haskell wiki、LYAH 和 Real World Haskell。而且难度简单反而对于没有学过任何函数式语言的人来说是好事。

**Comment by Eason2048** [comment]
Monad / Applicative这两章的exercise真难啊，想一整天也想不出怎么写。

**Reply by MTDickens** [reply]
我认为 Haskell MOOC 这一部分教的不够好，作业写对也是蒙对的，不知其所以然。可以暂时不写，而是去看看其他关于 monad 的 tutorials，比如

1. https://learnyouahaskell.com/functors-applicative-functors-and-monoids
2. https://wiki.haskell.org/All_About_Monads

等等。

或者这部分就先放一放，然后自己去看一看各种 Haskell 的实际代码，从各种代码中逐渐领悟 monad 的用法意义和好处。

**Reply by MTDickens** [reply]
这部分作业一定不要硬做，因为真的做完了也记不住什么。

**Reply by Eason2048** [reply]
https://sxyz.blog/functors-applicatives-and-monads-in-pictures/ 这个讲Monad/Applicative的不错

**Reply by Gravifer** [reply]
也推荐看看 CS3110 的 [§7.8.5](https://cs3110.github.io/textbook/chapters/conc/monads.html#monad-laws)

**Comment by Gravifer** [comment]
学完这门课后值得学习 Cornell CS3110 以获得[更大的视野](https://github.com/PKUFlyingPig/cs-self-learning/discussions/491#discussioncomment-6818625)； 后者包含了一些本课程中没有着重探索的内容；从我的视角来看，尤其是相较 Haskell mooc 多出的大量关于函数式数据结构的讨论非常有价值

**Comment by Garhlz** [comment]
机翻+llm润色了一下：
https://garhlz.github.io/haskell-mooc/
